core/Plugin/Report.php

```diff
@@ -848,17 +848,17 @@ public function getDimensions()
             $tableDimensionId = $this->getDimension()->getId();
             $dimensionId = str_replace('.', '_', $tableDimensionId);
             $dimensions[$dimensionId] = $this->getDimension()->getName();
-        }
 
-        for ($level = 1; $level <= 100; $level++) {
-            $subDimension = $this->getNthLevelTableDimension($level);
-            if (empty($subDimension) || $tableDimensionId === $subDimension->getId()) {
-                break;
-            }
-            $tableDimensionId = $subDimension->getId();
-            $subDimensionId = str_replace('.', '_', $tableDimensionId);
-            $dimensions[$subDimensionId] = $subDimension->getName();
-        };
+            for ($level = 1; $level <= 100; $level++) {
```

Comment:
Might be better to simply set `$tableDimensionId = null` in the beginning of the method, instead of moving the loop inside the if. That way we would cover cases, where the main table wouldn't have a dimension, but sub reports would have. Not sure how realistic that is, but won't hurt to support it.